llvm-external-projects/iree-dialects/lib/Dialect/LinalgTransform/IR/StructuredTransformOpsExt.cpp

```diff
@@ -128,11 +128,24 @@ static LogicalResult isEquivalentToOpImpl(PatternRewriter &rewriter,
                                           linalg::LinalgOp linalgOp,
                                           linalg::LinalgOp linalgModelOp) {
   // If basic properties do not match, return failure.
-  if (linalgOp.getInputs() != linalgModelOp.getInputs() ||
-      linalgOp.getOutputs() != linalgModelOp.getOutputs() ||
-      linalgOp.getIndexingMaps() != linalgModelOp.getIndexingMaps() ||
-      linalgOp.getIteratorTypesArray() != linalgModelOp.getIteratorTypesArray())
-    return failure();
+  {
+    linalg::OpOperandVector opInputs = linalgOp.getInputOperands();
+    linalg::OpOperandVector modelInputs = linalgModelOp.getInputOperands();
+    linalg::OpOperandVector opOutputs = linalgOp.getOutputOperands();
+    linalg::OpOperandVector modelOutputs = linalgModelOp.getOutputOperands();
+    auto notEqualFn = [](std::tuple<OpOperand *, OpOperand *> in) -> bool {
```

Comment:
Nit: I think this function is not needed. Equality of `SmallVector<T>` already compares the values of each elements.

Reply:
You should be checking the equality of values, not pointers.

Reply:
I guess I am not doing it right now. I should.